pkg/actors/internal_actor.go

```diff
@@ -112,31 +112,33 @@ func (c *internalActorChannel) InvokeMethod(ctx context.Context, req *invokev1.I
 		}
 		methodName := methodURL[methodStartIndex+len("/method/"):]
 
-		var requestData []byte
-		requestData, err = req.RawDataFull()
-		if err != nil {
-			return nil, err
-		}
-
 		// Check for well-known method names; otherwise, just call InvokeMethod on the internal actor.
 		if strings.HasPrefix(methodName, "remind/") {
 			reminderName := strings.TrimPrefix(methodName, "remind/")
-			reminderInfo := new(ReminderResponse)
-			if err = json.Unmarshal(requestData, reminderInfo); err != nil {
-				return nil, fmt.Errorf("unexpected ReminderResponse JSON payload: %s", requestData)
+			reminderInfo, ok := req.GetDataObject().(*ReminderResponse)
```

Comment:
This is needed to make tests pass.

Also avoids the unnecessary step of JSON marshalling and unmarshalling